src/object.d

```diff
@@ -3241,6 +3241,226 @@ template RTInfo(T)
     enum RTInfo = null;
 }
 
+int _adCmpT(T1, T2)(T1[] s1, T2[] s2)
+{
+    // U1 gets the unqualified version of T1
+    static if (is(T1 W == immutable W)) alias U1 = W;
+    else static if (is(T1 W == const W)) alias U1 = W;
+    else alias U1 = T1;
+
+    // U2 gets the unqualified version of T2
+    static if (is(T2 U == immutable U)) alias U2 = U;
+    else static if (is(T2 U == const U)) alias U2 = U;
+    else alias U2 = T2;
+
+    // objects
+    static if (is(U1 : Object) && is(U2 : Object))
+    {
+        auto c  = cast(sizediff_t)(s1.length - s2.length);
+        if (c == 0)
```

Comment:
I'm a bit confused here (I know it's the existing implementation, but just for my edification). This is not lexicographical compare because if an array is shorter than the other it always counts as smaller. For example, if we have an array `[hugeWidget]`, it always counts as smaller than `[littleWidget, littleWidget]`. Why is this the case?